ivy/functional/frontends/numpy/matrix/methods.py

```diff
@@ -23,13 +23,21 @@ def _init_data(self, data, dtype, copy):
                 dtype = data.dtype
             data = ivy.array(data, dtype=dtype, copy=copy)
             self._data = data
+        elif ivy.isscalar(data):
+            self._data = ivy.array(data)
```

Comment:
Can we use `ivy.asarray` then cast this scalar value to dtype

Reply:
done